### DISCUSSION TITLE
How to override column_type?

### DISCUSSION THREAD

**Discussion by lz1998** [SeaQL/sea-orm#473] [Q&A] [answered]
My database column is modified from `bigint` to `bigint unsigned`.

How to override column_type without modifying code from `i64` to `u64`? I know this column will not bigger than `i64::MAX`.

Code
```rust
#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Serialize, Deserialize)]
#[sea_orm(table_name = "xxx")]
pub struct Model {
    #[sea_orm(primary_key, auto_increment = false, column_type = "BigInteger")]
    pub id: i64,
    ....
}
```
Error
```bash
Err(Query Error: error occurred while decoding column "id": mismatched types; Rust type `core::option::Option<i64>` (as SQL type `BIGINT`) is not compatible with SQL type `BIGINT UNSIGNED`
```

**Comment by billy1624** [comment, marked as answer]
Hey @lz1998, I guess you are using PostgreSQL, try changing the `i64` to `u64`

**Reply by lz1998** [reply]
I am using MySQL

**Reply by billy1624** [reply]
Ok, it should be `u32` or `u64` since it's unsigned

**Reply by lz1998** [reply]
> Ok, it should be `u32` or `u64` since it's unsigned

Is it possible to cast u64 to i64 in Sea-ORM?

**Reply by billy1624** [reply]
This limitation comes from SQLx

**Reply by billy1624** [reply]
https://docs.rs/sqlx/0.5.10/sqlx/struct.MySql.html#impl-Type%3CMySql%3E-28
